### PR TITLE
ShareDrawer: Update documentation link buttons

### PR DESCRIPTION
**Pull request by juanicabanas**
**What is this feature?**

It updates the URL to get the correct documentation

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: public/app/features/dashboard/components/ShareModal/SharePublicDashboard/ModalAlerts/UnsupportedDataSourcesAlert.tsx [hunk 1]
```diff
@@ -33,7 +33,11 @@ export const UnsupportedDataSourcesAlert = ({ unsupportedDataSources }: { unsupp
         )}
       </p>
       <a
-        href="https://grafana.com/docs/grafana/next/dashboards/dashboard-public/"
+        href={
+          config.featureToggles.newDashboardSharingComponent
+            ? 'https://grafana.com/docs/grafana/next/dashboards/share-dashboards-panels/shared-dashboards/'
+            : 'https://grafana.com/docs/grafana/next/dashboards/dashboard-public/'
```
Comment (imatwawana) [inline public/app/features/dashboard/components/ShareModal/SharePublicDashboard/ModalAlerts/UnsupportedDataSourcesAlert.tsx:39]: Not sure about this link? `https://grafana.com/docs/grafana/next/dashboards/dashboard-public/ link`

**Review by imatwawana — COMMENTED** [review]
I'm not sure about one link

File: public/app/features/dashboard/components/ShareModal/SharePublicDashboard/ModalAlerts/UnsupportedDataSourcesAlert.tsx (hunk 1 above)
Comment (juanicabanas) [reply public/app/features/dashboard/components/ShareModal/SharePublicDashboard/ModalAlerts/UnsupportedDataSourcesAlert.tsx:39]: why?
I think, it should be 
`https://grafana.com/docs/grafana/next/dashboards/dashboard-public/#supported-data-sources` for the old design. What about the new one?

Comment (imatwawana) [reply public/app/features/dashboard/components/ShareModal/SharePublicDashboard/ModalAlerts/UnsupportedDataSourcesAlert.tsx:39]: I don't know what this component is, so I don't know why this link to the old docs is there. But as long as you're sure it should be there, then it's all good.

**Review by imatwawana — APPROVED** [review]

File: public/app/features/dashboard/components/ShareModal/SharePublicDashboard/ModalAlerts/UnsupportedDataSourcesAlert.tsx (hunk 1 above)
Comment (imatwawana) [reply public/app/features/dashboard/components/ShareModal/SharePublicDashboard/ModalAlerts/UnsupportedDataSourcesAlert.tsx:39]: Ah sorry, I misunderstood. It should be:
`https://grafana.com/docs/grafana/next/dashboards/share-dashboards-panels/shared-dashboards/#supported-data-sources`

**Pull request merged**